pkg/virt-controller/services/vm.go

```diff
@@ -59,49 +64,93 @@ func (v *vmService) DeleteVM(vm *corev1.VM) error {
 	return nil
 }
 
-func (v *vmService) GetRunningPods(vm *corev1.VM) (*v1.PodList, error) {
+func (v *vmService) GetRunningVMPods(vm *corev1.VM) (*v1.PodList, error) {
 	podList, err := v.KubeCli.Core().Pods(v1.NamespaceDefault).List(UnfinishedVMPodSelector(vm))
+
 	if err != nil {
 		return nil, err
 	}
 	return podList, nil
 }
 
+func UpdateMigration(migration *corev1.Migration) error {
+	restClient, err := kubecli.GetRESTClient()
+	if err != nil {
+		return err
+	}
+	migrationName := migration.ObjectMeta.Name
+	_, err = restClient.Put().Namespace(v1.NamespaceDefault).Resource("migrations").Body(migration).Name(migrationName).Do().Get()
+	return err
+}
+
+func GetDefinedVMs(vmName string) (*corev1.VM, error) {
```

Comment:
Looks like this can just return one VM.

Maybe `FetchVM`, would be ok too? The `Fetch` better indicates, that it is not just looked up in the cache ...